libsrc/apple2/stat.s

```diff
@@ -0,0 +1,163 @@
+;
+; Colin Leroy-Mira, 2023 <colin@colino.net>
+;
+; int __fastcall__ stat(const char *pathname, struct stat *statbuf);
+;
+
+        .export         _stat
+        .import         pushname, popname, _malloc, _free, _mktime, __errno
+        .import         prodos_datetime_to_posix_timespec, mli_file_info
+        .import         popax, pushax, pusha0, steaxspidx, incsp1, incsp2
+        .import         _open,_close
+        .include        "zeropage.inc"
+        .include        "errno.inc"
+        .include        "fcntl.inc"
+        .include        "filedes.inc"
+        .include        "mli.inc"
+        .include        "stat.inc"
+
+push_stbuf:
+        lda     stbuf
+        ldx     stbuf+1
+        jmp     pushax
+
+_stat:
+        ; Store statbuf
+        sta     ptr4
+        sta     stbuf
+        stx     ptr4+1
+        stx     stbuf+1
+
+        ; Clear statbuf
+        lda     #$00
+        ldy     #.sizeof(stat)-1
+:       sta     (ptr4),y
+        dey
+        bpl     :-
+
+        ; Reset errno
+        sta     ___errno
+
+        ; Store pathname
+        jsr     popax
+        sta     fname
+        stx     fname+1
+        jsr     pushax          ; Push it back for mli_file_info
+
+        jsr     mli_file_info
+        bcc     no_mli_err
+
+        jsr     popname         ; Preserves A and X
+        jmp     ___mappederrno
+
+no_mli_err:
+        ; st_dev
+        lda     DEVNUM
+        lsr                     ; Shift right to cc65 representation
+        lsr
+        lsr
+        lsr
+        ldy     #stat::st_dev
+        sta     (ptr4),y
+
+        ; st_access
+        lda     mliparam + MLI::INFO::ACCESS
+        ldy     #stat::st_access
+        sta     (ptr4),y
+
+        ; st_type and st_mode (S_IFDIR/S_IFREG only)
+        lda     mliparam + MLI::INFO::FILE_TYPE
+        ldy     #stat::st_type
+        sta     (ptr4),y
+        ldy     #stat::st_mode
+        cmp     #$0f
+        bne     is_reg
+        lda     #S_IFDIR
+        bne     set_st_mode
+
+is_reg: lda     #S_IFREG
+
+set_st_mode:
+        sta     (ptr4),y
+
+        ; st_auxtype
+        lda     mliparam + MLI::INFO::AUX_TYPE
+        ldy     #stat::st_auxtype
+        sta     (ptr4),y
+        lda     mliparam + MLI::INFO::AUX_TYPE + 1
```

Comment:
It's amazing how well you adapted to the existing coding style !!!

Just one minor thing here: When two "actual" values are added, then the `+` is separated with blanks. But if it's "just the usual + 1" (especially for 16-bit high-byte related stuff), then the blanks are omitted.